### PR TITLE
Correctly initialize the TabletType stats

### PR DESCRIPTION
**Pull request by dweitzman**
Fixes https://github.com/vitessio/vitess/issues/6988

Signed-off-by: David Weitzman <dweitzman@pinterest.com>

### CODE REVIEW

**Comment by dweitzman** [comment]
Converted to a draft because I noticed in testing that a tablet can still have the wrong `TabletType` in stats. I had a tablet started with type "spare" are converted to master that still describes itself as "spare".

**Comment by dweitzman** [comment]
Fixed to set the initial stat after checkMastership runs

File: go/vt/vttablet/tabletmanager/tm_init.go [hunk 1]
```diff
@@ -283,6 +283,7 @@ func (tm *TabletManager) Start(tablet *topodatapb.Tablet, healthCheckInterval ti
 
 	// The following initializations don't need to be done
 	// in any specific order.
+	statsTabletType.Set(topoproto.TabletTypeLString(tm.tmState.tablet.Type))
```
Comment (deepthi) [inline go/vt/vttablet/tabletmanager/tm_init.go:286]: Won't it be better to do this inside `exportStats`? Similar to how we are setting keyspace and shard stats.

Comment (dweitzman) [reply go/vt/vttablet/tabletmanager/tm_init.go]: That makes a lot of sense to me. Updated

**Review by sougou — APPROVED** [review]

**Comment by deepthi** [comment]
We should backport this to 7.0 and 8.0

**Pull request merged**